static/usage/modal/inline/basic/angular/app_component_ts.md

```diff
@@ -0,0 +1,29 @@
+```ts
+import { Component, ViewChild } from '@angular/core';
+import { IonModal } from '@ionic/angular';
+
+@Component({
+  selector: 'app-root',
+  templateUrl: 'app.component.html',
+})
+export class AppComponent {
+  @ViewChild(IonModal) modal: IonModal;
+
+  message = 'This modal example uses triggers to automatically open a modal when the button is clicked.';
+  name: string;
+
+  cancel() {
+    this.modal.dismiss(null, 'cancel');
+  }
+
+  confirm() {
+    this.modal.dismiss(this.name, 'confirm');
+  }
+
+  onDidDismiss(ev: any) {
```

Comment:
Can we use a stronger type? I believe the type here is `OverlayEventDetail<T>`

Reply:
This is due to this bug: https://github.com/ionic-team/ionic-framework/issues/24245. Do we want to show the workaround pattern instead here?

```ts
onDidDismiss(event: Event) {
  const ev = event as CustomEvent<OverlayEventDetail>;
}
```